stdlib/Profile/test/runtests.jl

```diff
@@ -272,12 +272,17 @@ end
 end
 
 @testset "HeapSnapshot" begin
-    fname = tempname()
-    run(`$(Base.julia_cmd()) --startup-file=no -e "using Profile; Profile.take_heap_snapshot($(repr(fname)))"`)
+    fname = strip(String(read(
+        `$(Base.julia_cmd()) --startup-file=no -E "using Profile; Profile.take_heap_snapshot()"`
+    )), ['\n', '\"'])
```

Comment:
You can't just randomly hack apart a Julia expression object and expect something meaningful back. Pass the thing you want directly:

```suggestion
    fname = read(`$(Base.julia_cmd()) --startup-file=no -e "using Profile; print(Profile.take_heap_snapshot())"`, String)
```

Reply:
makes sense. Thanks https://github.com/JuliaLang/julia/pull/47320